code/modules/awaymissions/maploader/reader.dm

```diff
@@ -349,11 +399,34 @@ var/global/dmm_suite/preloader/_preloader = new
 
 /dmm_suite/preloader/proc/load(atom/what)
 	for(var/attribute in attributes)
-		what.vars[attribute] = attributes[attribute]
+		var/value = attributes[attribute]
+		if(islist(value))
+			value = deepCopyList(value)
+		what.vars[attribute] = value
 	use_preloader = FALSE
 
+// A datum for use within the context of loading a single map,
+// so that one can have separate "unpowered" areas for ruins or whatever,
+// yet have a single area type for use of mapping, instead of creating
+// a new area type for each new ruin
+/dmm_suite/loaded_map
+	parent_type = /datum
+	var/list/area_list = list()
+
+/dmm_suite/loaded_map/proc/area_path_to_real_area(area/A)
+	if(!ispath(A, /area))
+		throw EXCEPTION("Wrong argument to `area_path_to_real_area`")
+
+	if(!(A in area_list))
+		if(initial(A.there_can_be_many))
+			area_list[A] = new A
+		else
+			area_list[A] = locate(A)
+
+	return area_list[A]
+
 /area/template_noop
```

Comment:
I'll highlight this since it might be interesting to -tg- devs - or they might realize this would just cause yet another slew of problems. What this does, is if the `/area` has `there_can_be_many` set to true, is create a distinct area for a loaded map, so that multiple space ruins maps can use the same `/area/ruin/unpowered` in their file, and not have APCs and air alarms behave linked together.